internal/persistence/entgo/postgres.go

```diff
@@ -29,40 +35,196 @@ import (
 
 	"github.com/siemens/wfx/generated/ent"
 	"github.com/siemens/wfx/persistence"
+
+	// AWS SDK for IAM authentication
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/feature/rds/auth"
 )
 
 //go:embed migrations/postgres/*.sql
 var postgresMigrations embed.FS
 
+const databaseConnectionTimeoutMilliseconds = 5000
+
 type PostgreSQL struct {
 	Database
 }
 
+// iamDB implements driver.Connector to support AWS RDS IAM authentication
+type iamDB struct {
+	config     *pgx.ConnConfig
+	awsConfig  aws.Config
+	useIAMAuth bool
+}
+
 func init() {
 	persistence.RegisterStorage("postgres", &PostgreSQL{})
 }
 
+// Connect implements driver.Connector interface
+// This method is called each time a new connection is needed, allowing for fresh IAM token generation
+func (id *iamDB) Connect(ctx context.Context) (driver.Conn, error) {
+	connConfig := id.config.Copy()
+
+	if id.useIAMAuth {
+		// Resolve the actual RDS endpoint (handles CNAME lookups)
+		actualHost, region, err := resolveRDSEndpoint(connConfig.Host)
```

Comment:
Why can't you just pass the correct endpoint to wfx? Why does wfx need so much logic to resolve the endpoint?